cli/daemon/dash/dashproxy/dashproxy.go

```diff
@@ -0,0 +1,53 @@
+// Package dashproxy proxies requests to the dash server,
+// caching them locally for offline access.
+package dashproxy
+
+import (
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"os"
+	"path/filepath"
+
+	"github.com/cockroachdb/errors"
+	"github.com/gregjones/httpcache"
+	"github.com/gregjones/httpcache/diskcache"
+	"github.com/peterbourgon/diskv"
+)
+
+func New(targetURL string) (*httputil.ReverseProxy, error) {
+	target, err := url.Parse(targetURL)
+	if err != nil {
+		return nil, errors.Wrap(err, "parse target url")
+	}
+
+	cacheDir, err := os.UserCacheDir()
+	if err != nil {
+		return nil, errors.Wrap(err, "get user cache dir")
+	}
+
+	cache := diskcache.NewWithDiskv(diskv.New(diskv.Options{
+		BasePath:     filepath.Join(cacheDir, "encore", "dashcache"),
+		CacheSizeMax: 1024 * 1024 * 1024, // 1GiB
+		Compression:  diskv.NewGzipCompression(),
+	}))
+	transport := httpcache.NewTransport(cache)
```

Comment:
When we deploy a new version of the dev dash to the website, how does the local cache know it needs to be cleared because there is a bug fix available online?

How do we make backwards incompatible changes to the dev dash without _breaking_ old installs? - Because if a users OS has caching disbaled or clears their cache - the next time they start up encore, we'll download an incompatible dashboard.

Maybe the URL needs to include the major Encore CLI version, so we can serve older versions if needed.